src/NuGetPackageVerifier/Rules/DependenciesVersionRule.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGetPackageVerifier.Rules
+{
+    public class DependenciesVersionRule : IPackageVerifierRule
+    {
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
+        {
+            if (!context.Metadata.Version.IsPrerelease)
+            {
+                foreach (var dependencyGroup in context.Metadata.DependencyGroups)
+                {                    
+                    foreach (var packageDependency in dependencyGroup.Packages)
+                    {
+                        if (packageDependency.VersionRange.HasUpperBound)
```

Comment:
Are these checks required prior to checking `MinVersion.IsPrerelease`? Or what happens if we omit this check?